src/robot/client.hpp

```diff
@@ -63,14 +63,26 @@ class RobotClient {
     /// Raises:
     /// 	Throws an error if the requested resource doesn't exist or is of the wrong type.
     ///
-    /// This function should not be called directly expect in specific cases. The
-    /// function `typed_resource_from_robot<T>(robot, name)` is the preferred method for obtaining
-    /// resources.
+    /// This method should not be called directly except in specific cases. The
+    /// type-annotated `resource_by_name<T>(name)` overload is the preferred method
+    /// for obtaining resources.
     ///
     /// Because the return type here is a `ResourceBase`, the user will need to manually
     /// cast to the desired type.
     std::shared_ptr<ResourceBase> resource_by_name(ResourceName name,
                                                    ResourceType type = {"resource"});
+    template <typename T>
+    std::shared_ptr<T> resource_by_name(const std::string& name) {
+        ResourceName r;
+        Subtype subtype = T::subtype();
+        *r.mutable_namespace_() = subtype.type_namespace();
+        *r.mutable_type() = subtype.resource_type();
+        *r.mutable_subtype() = subtype.resource_subtype();
+        *r.mutable_name() = std::move(name);
```

Comment:
Better, but you can't take `name` by `const` value either and get a `move` here, since a `const` object can't be moved from as that is a mutating operation.